WordPress/Classes/ViewRelated/NUX/SigninAccountAndBlogViewController.swift

```diff
@@ -0,0 +1,656 @@
+    /// - Parameters:
+    ///     - message: The message to display
+    ///
+    func displayLoginMessage(message: String!) {
+        statusLabel.text = message
+    }
+
+
+    /// Display an authentication status message
+    ///
+    /// - Parameters:
+    ///     - message: The message to display
+    ///
+    /// - Returns: the generated username
+    ///
+    func generateSiteTitleFromUsername(username: String) -> String {
+        // Currently, we set the title of a new site to the username of the account.
+        // Another possibility would be to name the site "username's blog", which is
+        // why this has been placed in a separate method.
+        return username
+    }
+
+
+    /// Displays an error message in an overlay
+    ///
+    /// - Parameters:
+    ///     - message: The message to display
+    ///
+    func displayErrorMessage(message: String) {
+        let presentingController = navigationController ?? self
+        let controller = SigninErrorViewController.controller()
+        controller.delegate = self
+        controller.presentFromController(presentingController)
+        controller.displayGenericErrorMessage(message)
+    }
+
+
+    /// Validates what is entered in the various form fields and, if valid,
+    /// proceeds with the submit action.
+    ///
+    func validateForm() {
+        view.endEditing(true)
+
+        // Is everything filled out?
+        if !SigninHelpers.validateFieldsPopulatedForCreateAccount(loginFields) {
+            displayErrorMessage(NSLocalizedString("Please fill out all the fields", comment: "A short prompt asking the user to properly fill out all login fields."))
+            return
+        }
+
+        if !SigninHelpers.validateFieldsForSigninContainNoSpaces(loginFields) {
+            displayErrorMessage(NSLocalizedString("Email, Username, and Site Address cannot contain spaces.", comment: "No spaces error message."))
+            return
+        }
+
+        if !SigninHelpers.validateUsernameIsMaxFiftyCharacters(loginFields.username) {
+            displayErrorMessage(NSLocalizedString("Username must be less than fifty characters.", comment: "Prompts that the username entered was too long."))
+            usernameField.becomeFirstResponder()
+            return
+        }
+
+        if !loginFields.emailAddress.isValidEmail() {
+            displayErrorMessage(NSLocalizedString("Please enter a valid email address", comment: "A short prompt asking the user to properly fill out all login fields."))
+            emailField.becomeFirstResponder()
+            return
+        }
+
+        // Remove ".wordpress.com" if it was entered.
+        loginFields.siteUrl = loginFields.siteUrl.componentsSeparatedByString(".")[0]
+
+        configureLoading(true)
+
+        createUserAndSite()
+    }
+
+
+    /// Composes and enqueues the operations to validate the site information,
+    /// user create, user authenticaiton, and blog creation.
+    ///
+    func createUserAndSite() {
```

Comment:
Hmm... I'm not sure I agree. We'd be creating a new service just to call other services and the additional abstraction will just make things more complicated.